sbi/simulators/simutils.py

```diff
@@ -150,25 +143,41 @@ def process_pytorch_prior(prior: Distribution,) -> Tuple[Distribution, int, bool
     return prior, theta_dim, False
 
 
-# XXX This is equally liable to happen with others, e.g. Beta...
-# XXX Wrapping with Independent seems to be a universally valid remedy
-def check_for_batch_reinterpretation_extra_d_uniform(prior):
-    """Raise ValueError in case of inadvertent use of batched scalar Uniform as prior.
+def maybe_reinterpret_batch_dims(prior) -> Distribution:
+    """Return prior with batch shape smaller equal one. 
+
+    Reinterprets all batch dimensions as event dimensions if possible, e.g., for
+    Uniform, Beta, Gamma priors. 
 
-    Most likely the user needs to specify a prior on a multi-dimensional theta
-    rather than several batched 1D priors at once.
+    Raises: 
+        ValueError: if batch_shape larger 1 and reinterpretation is not possible. 
+
+    Returns:
+        prior: prior with maybe reinterpreted batch dimensions.
     """
 
-    # Note .batch_shape will always work because and is short-circuiting.
-    if isinstance(prior, (Uniform, Beta)) and prior.batch_shape.numel() > 1:
-        raise ValueError(
+    reinterpretable_priors = (Uniform, Beta, Gamma)
+
+    if prior.batch_shape.numel() <= 1:
+        return prior
+
+    elif isinstance(prior, reinterpretable_priors) and prior.batch_shape.numel() > 1:
```

Comment:
Nicely done. Please disregard my previous comment (#135), except... do we not want to reinterpret always all batch dimensions? or only for distributions with `event_shape=1`?

I was under the impression that we never want batch dimensions.

Anyways, when the PR on MultipleIndependent comes in this point is moot.

Reply:
I think we dont to always reinterpret. Here we reinterpret because we think the user doesnt know what they are doing, i.e., creating batch shape >1 with event shape = 1, instead of event shape >1 (only because event shape cant be larger than 1 for that distribution). 

However, if event shape > 1, this means that the underlying distributions can in fact be multivariate and the user probably specified the batch shape on purpose.

Reply:
In any case, in the future we will throw a value error or so if batch_shape > 1 and kindly ask the user to specify univariate priors with event shape > 1 by passing an `IndependentJoint`.